lib/rewards/server/models/reward.js

```diff
@@ -0,0 +1,161 @@
+            return false;
+        }
+
+        return true;
+    }
+
+    isItemType()
+    {
+        return null !== this.itemId;
+    }
+
+    isModifierType()
+    {
+        return null !== this.modifierId;
+    }
+
+    hasExperienceSet()
+    {
+        return 0 <= this.experience;
+    }
+
+    hasAnimationData()
+    {
+        return this.hasDropBody && sc.get(this, 'animationData', false);
+    }
+
+    static areValidRewards(rewards)
+    {
+        return 0 < (rewards?.length || 0);
+    }
+
+    static getRewardsBag(rewards)
+    {
+        let itemRangeArray = [];
+
+        for (let reward of rewards) {
+            let itemRangeCount = itemRangeArray.length;
+            for (let i = 0; i < reward.dropRate; i++) {
+                itemRangeArray[itemRangeCount + i] = reward;
+            }
+        }
+
+        return itemRangeArray;
+    }
+
+    static fromModel(rewardModel)
+    {
+        if (!rewardModel) {
+            return null;
+        }
+        return new Reward({
+            'id': rewardModel.id,
+            'objectId': rewardModel.object_id,
+            'itemId': rewardModel.item_id,
+            'modifierId': rewardModel.modifier_id,
+            'experience': rewardModel.experience,
+            'dropRate': rewardModel.drop_rate,
+            'dropQuantity': rewardModel.drop_quantity,
+            'isUnique': rewardModel.is_unique,
+            'wasGiven': rewardModel.was_given,
+            'hasDropBody': rewardModel.has_drop_body,
+            'animationData': ObjectsItemsRewardsAnimations.fromModel(rewardModel.objects_items_rewards_animations)
+        });
+    }
+
+    static createDropObjectData(reward, roomId)
+    {
+        return {
+            id: reward.randomRewardId+reward.tileIndex,
+            room_id: roomId,
+            layer_name: reward.randomRewardId,
+            tile_index: reward.tileIndex,
+            object_class_key: ObjectsConst.TYPE_DROP,
+            client_key: reward.randomRewardId+reward.tileIndex,
+            rewardModel: reward,
+            asset_key: reward.animationData.assetKey,
+            client_params: JSON.stringify({
+                frameStart: reward.animationData.extraParams.start,
+                frameEnd: reward.animationData.extraParams.end,
+                repeat: -1,
+                hideOnComplete: false,
+                autoStart: true,
+                asset_key: reward.animationData.assetKey
+            }),
+            enabled: 1,
+            objects_assets: [{
+                object_asset_id: undefined,
+                object_id: reward.randomRewardId,
+                asset_type: reward.animationData.assetType,
+                asset_key: reward.animationData.assetKey,
+                file_1: reward.animationData.file,
+                file_2: null,
+                extra_params: `{"frameWidth":${reward.animationData.extraParams.frameWidth},"frameHeight":${reward.animationData.extraParams.frameHeight}}`
```

Comment:
JSON.stringify(reward.animationData.extraParams) should set all the params at once, and don't need to create this string manually

Reply:
can't just pass or "map" the entire `reward.animationData.extraParams` into the expected `client_params`?

Reply:
1) The are params inside the client_params that don't come from animationData.extra_params, like ```hideOnComplete```, ```autoStart``` or  ```asset_key```.
2) No, because someone down the line creating the object inside the objectManager is expecting that to be a JSON string, not an actual object.